### ISSUE TITLE
commit 8005ca4 "Move strlcat, strlcpy, and strnlen"  is not working as it suposed to be

### ISSUE THREAD

**Issue by BrainSlayer** [zfsonlinux/zfs#8157]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| OpenSuse
Distribution Version	| 15.0
Linux Kernel	| 4.15
Architecture	| x64
ZFS Version	| master
SPL Version	| master

### Describe the problem you're observing
the commit 8005ca4 "Move strlcat, strlcpy, and strnlen"  is not working as it suposed to be since it is overriding existing declarations from glibc and does break compiling with musl. it should not replace strlcpy / strlcat with own implementations but using standard glibc functions instead (if available)

### Describe how to reproduce the problem
for checking if it breaks with musl. just compile it with musl library. alpine linux for instance is based on musl. i tested it with cross compilers on my embedded system targets

**Comment by BrainSlayer** [comment]
please read https://github.com/zfsonlinux/zfs/commit/8005ca4f749f71397197292452ca539a25286b89 for more informations and complaints of other users

**Comment by MyPod** [comment]
I've attempted to build the recently-released 0.8.0-rc3 on Debian stable and am hitting what looks related issues.

When running any sort of packaging action (make pkg-utils|pkg-kmod) I get the following output:

```make[4]: Entering directory '/root/zfs/lib/libspl'
make[4]: *** No rule to make target 'strnlen.c', needed by 'distdir'.  Stop.
make[4]: Leaving directory '/root/zfs/lib/libspl'
Makefile:705: recipe for target 'distdir' failed
make[3]: *** [distdir] Error 1
make[3]: Leaving directory '/root/zfs/lib'
Makefile:883: recipe for target 'distdir' failed
make[2]: *** [distdir] Error 1
make[2]: Leaving directory '/root/zfs'
Makefile:985: recipe for target 'dist' failed
make[1]: *** [dist] Error 2
make[1]: Leaving directory '/root/zfs'
Makefile:1221: recipe for target 'srpm-utils' failed
make: *** [srpm-utils] Error 2```

**Comment by MyPod** [comment]
Sorry for the noise, it seems I forgot a `./autogen.sh`.
